src/components/ValuePicker/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React, {forwardRef, useState} from 'react';
+import type {ForwardedRef} from 'react';
+import {View} from 'react-native';
+import FormHelpMessage from '@components/FormHelpMessage';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import type {ValuePickerItem, ValuePickerProps} from './types';
+import ValueSelectorModal from './ValueSelectorModal';
+
+function ValuePicker({value, label, items, placeholder = '', errorText = '', onInputChange, furtherDetails, shouldShowTooltips = true}: ValuePickerProps, forwardedRef: ForwardedRef<View>) {
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
+
+    const showPickerModal = () => {
+        setIsPickerVisible(true);
+    };
+
+    const hidePickerModal = () => {
+        setIsPickerVisible(false);
+    };
+
+    const updateInput = (item: ValuePickerItem) => {
+        if (item?.value && item.value !== value) {
+            onInputChange?.(item.value);
```

Comment:
Maybe it will be more safe to update `onInputChange` typing then adding an extra check `item?.value &&` .

Reply:
Tbh I don't know what you mean 😄 Can you explain?

Reply:
I mean in old code if `item.value` was `undefined`  it still was possible for it to be passed to `onInputChange`.
Right now we updated the if condition that changed it. I'm not sure if there can be a real case where item.value is undefined.
But to escape unexpected situation, we can turn the check as it was before
```suggestion
       if (item.value !== value) {
            onInputChange?.(item.value);
```
and just update the `onInputChange` type instead
```
 onInputChange?: (value: string | undefined) => void;
```

Reply:
Done